### PR TITLE
Update package structure

### PR DESCRIPTION
**Pull request by fanatid**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
+  "author": "T. Jameson Little <t.jameson.little@gmail.com>",
+  "files": [
+    "index.js",
+    "base64js.min.js"
+  ],
+  "main": "index.js",
   "repository": {
     "type": "git",
-    "url": "git://github.com/beatgammit/base64-js.git"
+    "url": "https://github.com/beatgammit/base64-js.git"
```
Comment (feross) [inline package.json]: This should remain as git://

File: package.json [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
+  "homepage": "https://github.com/beatgammit/base64-js",
+  "bugs": {
+    "url": "https://github.com/beatgammit/base64-js/issues"
+  },
+  "license": "MIT",
+  "author": "T. Jameson Little <t.jameson.little@gmail.com>",
+  "files": [
+    "index.js",
+    "base64js.min.js"
+  ],
```
Comment (feross) [inline package.json]: Not a fan of the `"files"` field. Tests should be shipped on npm.

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,17 @@
...
+  - "6"
+env:
+  matrix:
+    - TEST_SUITE=unit
+matrix:
+  include:
+    - os: linux
+      node_js: "6"
+      env: TEST_SUITE=lint
+script: npm run $TEST_SUITE
```
Comment (feross) [inline .travis.yml]: What does all this do? Wasn't the file perfectly fine before?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+node_modules
+
```
Comment (feross) [inline .gitignore]: Remove newline

File: .npmignore [hunk 5]
```diff
@@ -1 +0,0 @@
-bench
```
Comment (feross) [inline .npmignore]: Please add this file back and remove the "files" field from package.json

**Comment by feross** [comment]
Overall, good changes, but please address the inline comment.

File: .npmignore (hunk 5 above)
Comment (fanatid) [inline .npmignore]: @feross are `.npmignore` better for you than `files`? I always thought that white list is better.

Comment (feross) [inline .npmignore]: Yeah, when it comes to security-related things, you're right whitelists are better. I just think that tests are part of the documentation for the package and should be published to npm. So it's easier to just ignore the `bench` folder than to whitelist every file.

Comment (fanatid) [inline .npmignore]: ok, why not add `tests` to `files` then?

File: .travis.yml (hunk 3 above)
Comment (fanatid) [inline .travis.yml]: old config run `lint` and `unit` for every version
new run `unit` for every version and `lint` only for node 6 (so now it's more easy determine what's wrong: unit or lint)

Comment (feross) [inline .travis.yml]: Seems complicated, `standard` runs in < 5 seconds and you can just look at the error output to figure out what is wrong. But I guess this is fine...

File: .npmignore (hunk 5 above)
Comment (feross) [inline .npmignore]: Then you have to remember to add new test files to the `files` field. We're not putting secrets or password files in this repo, so there's no real risk of accidentally publishing something sensitive so the blacklist (npmignore) approach is actually simpler IMO

Comment (fanatid) [inline .npmignore]: We need only add folder `test`.
But ok, I'll change back or we create new holywar thread about files/npmignore :grinning:

Comment (feross) [inline .npmignore]: Adding `test` is fine. I thought you need to add each individual file.

File: .travis.yml (hunk 3 above)
Comment (fanatid) [inline .travis.yml]: It is not complicated, please check [travis build](https://travis-ci.org/beatgammit/base64-js/builds/161847211) and maybe you will agree that new config is good.

Comment (feross) [inline .travis.yml]: Okay, that's actually really cool! 👍

File: package.json (hunk 1 above)
Comment (fanatid) [inline package.json]: done

File: .travis.yml [hunk 6]
```diff
@@ -1,6 +1,17 @@
+sudo: false
+os:
+  - linux
```
Comment (feross) [inline .travis.yml]: Can this line be removed? I've always gotten linux test machines when it's not specified.

File: .gitignore (hunk 4 above)
Comment (fanatid) [inline .gitignore]: removed

File: .travis.yml [hunk 7]
```diff
@@ -1,6 +1,17 @@
...
 language: node_js
 node_js:
-  - "0.10"
   - "0.12"
   - "4"
-  - "node"
+  - "6"
+env:
+  matrix:
```
Comment (feross) [inline .travis.yml]: What does `matrix` mean?

File: .travis.yml (hunk 6 above)
Comment (fanatid) [inline .travis.yml]: I think yes, but we need same line in `matrix` AFAIR, so I prefer hold `os` in main section too.

File: .travis.yml (hunk 7 above)
Comment (fanatid) [inline .travis.yml]: This option allows combine all possible combinations of options. [docs on travis](https://docs.travis-ci.com/user/customizing-the-build#Build-Matrix)

**Comment by feross** [comment]
Can you rebase? Then I can merge.

File: .npmignore (hunk 5 above)
Comment (fanatid) [inline .npmignore]: Added `test` to `files`, is it ok? or return to `.npmignore` will be better?

File: .travis.yml (hunk 6 above)
Comment (feross) [inline .travis.yml]: ok

**Comment by feross** [comment]
Released as 1.2.0

**Pull request merged**